django/test/utils.py

```diff
@@ -25,6 +25,7 @@
 from django.db import DEFAULT_DB_ALIAS, connections, reset_queries
 from django.db.models.options import Options
 from django.template import Template
+from django.template.defaulttags import TemplateProxy
```

Comment:
Shall we move the `TemplateProxy` to this file instead? :thought_balloon: 

I see there could be a circular dependency between defaulttags and utils, which feels like a code smell. Can you think of alternatives to clean defaulttags a bit and leave the code the more focused for the new tags and have helpers somewhere else?

Reply:
@nessita I'm not particularly fussed about the name. If you prefer `TemplatePartial` then ok, maybe. 🤷 (The original name was because it proxies for a template, providing the ability to bind the context to the node list when rendering) 

I'm sceptical about using inheritance here. Bar render then implementations aren't shared. (And render is very few lines.) We not specialising. Etc. (I.e. the positive reasons for subclassing aren't in play. All there is, is maybe save a few lines of typing, which isn't a good reason.)